protocol/xmrtaker/instance.go

```diff
@@ -271,26 +270,9 @@ func (inst *Instance) maybeCancelNewSwap(txHash ethcommon.Hash) (bool, error) {
 
 	// just double the gas price for now, this is higher than needed for a replacement tx though
 	gasPrice := new(big.Int).Mul(tx.GasPrice(), big.NewInt(2))
-	cancelTx, err := inst.backend.ETHClient().CancelTxWithNonce(inst.backend.Ctx(), tx.Nonce(), gasPrice)
+	receipt, err := inst.backend.ETHClient().CancelTxWithNonce(inst.backend.Ctx(), tx.Nonce(), gasPrice)
 	if err != nil {
-		return false, fmt.Errorf("failed to create or send cancel tx: %w", err)
-	}
-
-	log.Infof("submit cancel tx %s", cancelTx)
-
-	// TODO: if newSwap is included instead of the cancel tx (unlikely), block.WaitForReceipt will actually
-	// loop for 1 hour and block as there will never be a receipt for the cancel tx.
```

Comment:
I don't think this comment is correct, at least not on mainnet or sepolia, so I didn't move it to another point in the code. (Ganache doesn't have a mempool, and it might not handle things correctly.) Whichever transaction gets included in a block first will successfully get a receipt, while the other one will get an error waiting for a receipt because it uses the same nonce and can't be included in any future block.